ros2_socketcan/src/socket_can_receiver.cpp

```diff
@@ -171,18 +171,24 @@ CanId SocketCanReceiver::receive_fd(void * const data, const std::chrono::nanose
   // Read
   struct canfd_frame frame;
   const auto nbytes = read(m_file_descriptor, &frame, sizeof(frame));
+
   // Checks
   if (nbytes < 0) {
     throw std::runtime_error{strerror(errno)};
   }
```

Comment:
```suggestion
  }

  if (frame.len > CANFD_MAX_DLEN) {
    throw std::runtime_error{"read: frame length is larger than max allowed CAN FD payload length"};
  }
```

> ```
> /* CAN FD payload length and DLC definitions according to ISO 11898-7 */
> #define CANFD_MAX_DLC 15
> #define CANFD_MAX_DLEN 64
> ```

https://github.com/linux-can/can-utils/blob/6526683a10d8700ceea5148e5d2652fadaf3db6d/include/linux/can.h#L93-L95

Since now we are doing dynamic size checking, we can add additional safety.